### ISSUE TITLE
Freezes with screen saying "Load Next Game"

### ISSUE THREAD

**Issue by ukscone** [filmote/LodeRunner#9]
I've tried building from source and also the released .hex files and all that happens on my arduboy (arduventure se) is it just sits there with the message on screen of line of dots, load next game and a line of dots.

any ideas?

**Comment by ukscone** [comment]
never mind I worked out the problem. well how to solve it anyway. had to use arduboy assistant to clear the eeprom and everything seems to be fine.

**Comment by filmote** [comment]
Really, had you been playing the game prior to this?  I have a check in the startup to see if the EEPROM has been initialised before playing.

**Comment by dogeared** [comment]
I am seeing this same issue. I have not played the game before. I originally tried to install the game from the arduino IDE. That's when I first saw the message. 

I subsequently loaded the hex files using `avrdude` on my mac, but I still see that message.

If clearing the EEPROM is the issue, how does one go about doing that?

Thanks!

**Comment by dogeared** [comment]
Success!  I was able to clear the EEPROM using the https://community.arduboy.com/t/arduboy-assistant-handy-arduboy-management-tool/2096 tool.

I'd like to write a readme as well as an install script, if you're interested in the contribution.

Evade 2 - https://github.com/ModusCreateOrg/evade2 - has an awesome install script that works on all the platforms.

**Comment by Pharap** [comment]
It doesn't really need an install script, that's a bit overkill.
Evade2 has an install script because they removed the USB code so it doesn't work very well with the normal install/uninstall methods.

The root of the problem here is that there's some game that's overwriting LodeRunner's save area and leaving it in a state where LodeRunner doesn't realise that it wasn't the last game to write to that area.
All LodeRunner needs is a way to clear the EEPROM,
and maybe a better way to detect whether the memory is valid or not as a precaution.

**Comment by dogeared** [comment]
fair enough. I do think it would benefit from a readme. As a total n00b and running on mac, there really wasn't any way to know the best way of installing the game and ensuring that it could run. It took me a while to realize I didn't need to build the app at all and that I could just install the `.hex` files. Then, once I realized that, it took me a while to find out how to get `avrdude` working properly.

The only thing that I'd had on the arduboy was what it shipped with (I forget) and the HelloWorld example from the tutorial.

Then, I loaded up LodeRunner in the Arduino IDE and installed it to the arduboy. Immediately, I got the message: `Load next game`.

Without knowing 100%, I think installing LodeRunner the way I did maybe is what had the wrong info in the eeprom? For instance, how *would* one install the first batch of levels from source? That could be in the readme too.

At any rate - thanks for building this! Very nostalgic for me. I'm on level 5 now. 145 to go! ;)

**Comment by Pharap** [comment]
The reason there isn't a guide for how to compile/install is because it uses the same mechanism all games use.
The standard procedure is to take the source and build it on the Arduino IDE, which is fairly easy once you get used to it.
Some people prefer to use one of the various 'loader' programs available to load `.hex` files.
Of those, avrdude is probably the most fiddly to get working because it's console based and more specific about its timing.

What is different though is that the game is split over several `.hex` files (a bit like how really old games were stored on multiple floppy disks).
I think it's fair to say it could at least do with a link to the thread that explains the multiple game file mechanism, and/or have the same information duplicated in the readme.

The EEPROM issue is more likely coming from a conflict with another game.
EEPROM generally isn't altered when a game is uploaded.
There's only so much EEPROM (1024 bytes), and all games have to share it,
so different people come up with different mechanisms to try to figure out whether the area their games use has been corrupted or not,
and unfortunately the mechnism LodeRunner currently uses isn't quite foolproof.
